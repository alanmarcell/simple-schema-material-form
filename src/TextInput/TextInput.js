import React from 'react';
import PropTypes from 'prop-types';

const TextInput = props => {
  console.log('TextInput >>>', props);
  return <input {...props} />;
};

TextInput.defaultProps = {
  value: '',
};

TextInput.propTypes = {
  value: PropTypes.string,
  doc: PropTypes.object.isRequired,
  setDoc: PropTypes.func.isRequired,
};

export default TextInput;
