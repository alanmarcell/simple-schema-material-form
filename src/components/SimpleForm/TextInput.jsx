import React from 'react';
// import PropTypes from 'prop-types';
import { omit, pathOr } from 'ramda';
// import { compose, withStateHandlers, pure } from 'recompact';
import { TextField } from 'material-ui';


const TextInput = props => {
  const { doc, setDoc, fieldName } = props;
  const value = pathOr('', [fieldName], doc);
  const cProps = omit(['fieldName', 'setDoc', 'doc'], props);

  const inputProps = {
    ...cProps,
    onChange: e => setDoc({ [fieldName]: e.target.value }),
    value,
    label: fieldName,
  };
  return <TextField {...inputProps} />;
};

export default TextInput;
