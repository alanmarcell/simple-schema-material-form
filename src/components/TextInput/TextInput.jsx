import React from 'react';
import PropTypes from 'prop-types';
import { TextField } from 'material-ui';
import { withStateHandlers, compose } from 'recompact';

/**
 * Component is described here.
 *
 */
const TextInput = ({ value, onChange }) => (
  <div>
    <TextField type="text" label={value} value={value} onChange={onChange} />
    <p>
      {value}
    </p>
  </div>
);

const enhance = compose(withStateHandlers(
  { value: '' },
  {
    onChange: () => e => ({
      value: e.target.value,
    }),
  },
));

const EnhancedTextInput = enhance(TextInput);


TextInput.defaultProps = {
  value: 'Initial',
};

TextInput.propTypes = {
  value: PropTypes.string,
};

export default EnhancedTextInput;
