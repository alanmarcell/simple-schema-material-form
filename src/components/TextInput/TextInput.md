# TextField With Recompose:

```jsx
const Values = require('./Values').default;
props = { fieldName : 'alan', value: 'Initial Value' };
initialState = { value: props.value, fieldName: props.fieldName};
<div>
  <Values {...state} />
  <TextInput
    {...props}
    onChange={e => { setState({value: e.target.value}); props.updateStateValue(e.target.value)}}
    ref={ref =>  (props = ref)} />
  <TextInput fieldName="thais" />
</div>
```
