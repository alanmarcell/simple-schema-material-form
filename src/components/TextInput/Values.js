import React, { Component } from 'react';
import { mapObjIndexed, values } from 'ramda';

const ValuesList = props => {
  return (
    <div key={props.propName} >
      <p>{props.propName}: {props.propValue}</p>
    </div>
  );
};

export default class Values extends Component {
  render() {
    const propValues = values(mapObjIndexed((val, key) => ({ propName: key, propValue: val }), this.props));

    return (
      <div>
        <div>{propValues.map(ValuesList)}</div>
      </div>
    );
  }
}
