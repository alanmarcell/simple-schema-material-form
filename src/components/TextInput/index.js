import { compose, withState } from 'recompose';

import TextInputComponent from './TextInput';

const enhance = compose(withState('value', 'setvalue', ({ value }) => value));

export { TextInputComponent, enhance };

export default TextInputComponent;
