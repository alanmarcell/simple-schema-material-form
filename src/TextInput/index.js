import { pathOr } from 'ramda';
import { compose, pure, setDisplayName, mapProps } from 'recompose';
import TextInput from './TextInput';

// const enhance = compose(withState('value', 'setvalue', ({ value }) => value));

// export { TextInputComponent, enhance };

const enhance = compose(
  pure,
  setDisplayName('TextInput'),
  mapProps(({
    fieldName, doc, setDoc, ...props
  }) => ({
    label: fieldName,
    value: pathOr('', [fieldName], doc),
    onChange: e => setDoc({ [fieldName]: e.target.value }),
    ...props,
  })),
);

const EnhancedTextInput = enhance(TextInput);

export default EnhancedTextInput;
