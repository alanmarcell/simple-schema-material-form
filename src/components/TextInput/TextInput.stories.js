import React from 'react';
import { storiesOf } from '@storybook/react';
import { withState } from '@dump247/storybook-state';
import withTests from '../../stories/withTests';
import TextInput from './';

storiesOf('Fields|TextInput', module)
  .addDecorator(withTests('TextInput'))
  .add('simple', withState({ doc: {} }, (store) => (
    <TextInput
      {...store.state}
      setDoc={e => store.set({ doc: e })}
      fieldName="World"
    />
  )));

