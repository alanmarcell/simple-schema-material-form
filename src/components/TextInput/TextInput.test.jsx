import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import TextInput from './TextInput';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TextInput fieldName="World" />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  const component = shallow(<TextInput fieldName="World" />);
  component.dive();
  expect(component).toMatchSnapshot();
});
