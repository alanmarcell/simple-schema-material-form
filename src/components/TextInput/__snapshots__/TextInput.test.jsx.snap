// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withStateHandlers(TextInput)
    fieldName="World"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TextField
          label=""
          onChange={[Function]}
          required={false}
          select={false}
          type="text"
          value=""
/>,
        <p />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "",
          "onChange": [Function],
          "required": false,
          "select": false,
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TextField
            label=""
            onChange={[Function]}
            required={false}
            select={false}
            type="text"
            value=""
/>,
          <p />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "",
            "onChange": [Function],
            "required": false,
            "select": false,
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
